daemon/networkdriver/bridge/driver.go

```diff
@@ -513,70 +506,68 @@ func linkLocalIPv6FromMac(mac string) (string, error) {
 }
 
 // Allocate a network interface
-func Allocate(job *engine.Job) error {
+func Allocate(id, requestedMac, requestedIP, requestedIPv6 string) (*network.Settings, error) {
 	var (
-		ip            net.IP
-		mac           net.HardwareAddr
-		err           error
-		id            = job.Args[0]
-		requestedIP   = net.ParseIP(job.Getenv("RequestedIP"))
-		requestedIPv6 = net.ParseIP(job.Getenv("RequestedIPv6"))
-		globalIPv6    net.IP
+		ip         net.IP
+		mac        net.HardwareAddr
+		err        error
+		globalIPv6 net.IP
 	)
 
-	ip, err = ipAllocator.RequestIP(bridgeIPv4Network, requestedIP)
+	ip, err = ipAllocator.RequestIP(bridgeIPv4Network, net.ParseIP(requestedIP))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// If no explicit mac address was given, generate a random one.
-	if mac, err = net.ParseMAC(job.Getenv("RequestedMac")); err != nil {
+	if mac, err = net.ParseMAC(requestedMac); err != nil {
 		mac = generateMacAddr(ip)
 	}
 
 	if globalIPv6Network != nil {
 		// If globalIPv6Network Size is at least a /80 subnet generate IPv6 address from MAC address
 		netmaskOnes, _ := globalIPv6Network.Mask.Size()
-		if requestedIPv6 == nil && netmaskOnes <= 80 {
-			requestedIPv6 = make(net.IP, len(globalIPv6Network.IP))
-			copy(requestedIPv6, globalIPv6Network.IP)
+		ipv6 := net.ParseIP(requestedIPv6)
+		if ipv6 == nil && netmaskOnes <= 80 {
+			ipv6 = make(net.IP, len(globalIPv6Network.IP))
+			copy(ipv6, globalIPv6Network.IP)
 			for i, h := range mac {
-				requestedIPv6[i+10] = h
+				ipv6[i+10] = h
 			}
 		}
 
-		globalIPv6, err = ipAllocator.RequestIP(globalIPv6Network, requestedIPv6)
+		globalIPv6, err = ipAllocator.RequestIP(globalIPv6Network, ipv6)
 		if err != nil {
 			logrus.Errorf("Allocator: RequestIP v6: %v", err)
-			return err
+			return nil, err
 		}
 		logrus.Infof("Allocated IPv6 %s", globalIPv6)
 	}
 
-	out := engine.Env{}
-	out.Set("IP", ip.String())
-	out.Set("Mask", bridgeIPv4Network.Mask.String())
-	out.Set("Gateway", bridgeIPv4Network.IP.String())
-	out.Set("MacAddress", mac.String())
-	out.Set("Bridge", bridgeIface)
-
-	size, _ := bridgeIPv4Network.Mask.Size()
-	out.SetInt("IPPrefixLen", size)
+	maskSize, _ := bridgeIPv4Network.Mask.Size()
 
 	// If linklocal IPv6
 	localIPv6Net, err := linkLocalIPv6FromMac(mac.String())
 	if err != nil {
-		return err
+		return nil, err
 	}
 	localIPv6, _, _ := net.ParseCIDR(localIPv6Net)
-	out.Set("LinkLocalIPv6", localIPv6.String())
-	out.Set("MacAddress", mac.String())
+
+	networkSettings := &network.Settings{
+		IPAddress: ip.String(),
+		//Mask:                 bridgeIPv4Network.Mask.String(),
```

Comment:
yes, to bring up your attention :P

Basically, it was set here https://github.com/docker/docker/pull/12092/files#diff-fb93346b0b06f64fcb85ea4a7de12f8dL558 but apparently never used. So just wanted to make sure it's normal and okay to remove.